Week1/homework/app.js

```diff
@@ -1,11 +1,114 @@
+
+    div.appendChild(ul);
+
+    for (let k = 0; k < myArr.length; k++) {
+      let li = document.createElement('li');
+      ul.appendChild(li);
+      li.innerHTML = myArr[k];
+    }
+  }
+
+  printBookTitles("div1", bookTitles);
+  */
+
+  const bookShelf = {
+    "crime_and_punisment": { title: "Crime and Punisment", author: "Dostoyevski", language: "Russian", publisher: "Oxford", "first edition": 1895 },
+    "zorba": { title: "Zorba", author: "Nikos Kazancakis", language: "Greek", publisher: "Can", "first edition": 1955 },
+    "siddhartha": { title: "Siddhartha", author: "Hermann Hesse", language: "German", publisher: "Inkilap", "first edition": 1895 },
+    "white_fang": { title: "White Fang", author: "Jack London", language: "English", publisher: "Isbankasi", "first edition": 1905 },
+    "poor_folk": { title: "Poor Folk", author: "Dostoyevski", language: "Russian", publisher: "YKK", "first edition": 1867 },
+    "white_nights": { title: "White Nights", author: "Dostoyevski", language: "Russian", publisher: "Kobo Co", "first edition": 1854 },
+    "sorrows_young_werther": { title: "The Sorrows of Young Werther", author: "Goethe", language: "German", publisher: "Bordo&Siyah", "first edition": 1882 },
+    "sweet_orange_tree": { title: "My Sweet Orange Tree", author: "Vasconcelos", language: "Portuguese", publisher: "Can Publisher", "first edition": 1986 },
+    "what_men_live_by": { title: "What Men Live By?", author: "Tolstoy", language: "Russian", publisher: "Oxford", "first edition": 1878 },
+    "the_alchemist": { title: "The Alchemist", author: "Paulo Coelho", language: "Portuguese", publisher: "Metis", "first edition": 2005 }
+  };
+
+  const bookCover = {
+    "crime_and_punisment": "https://is1-ssl.mzstatic.com/image/thumb/Publication/v4/c7/5f/93/c75f93e2-9f03-82a0-3970-332d46ce3b26/source/1200x630bb.jpg",
+    "zorba": "https://images-na.ssl-images-amazon.com/images/I/81znZ1KoT6L._SY445_.jpg",
+    "siddhartha": "https://images-na.ssl-images-amazon.com/images/I/51giy3zae6L._SX329_BO1,204,203,200_.jpg",
+    "white_fang": "https://images-na.ssl-images-amazon.com/images/I/41E5RNrSKhL._SX331_BO1,204,203,200_.jpg",
+    "poor_folk": "https://images-na.ssl-images-amazon.com/images/I/51iPBDEi1SL.jpg",
+    "white_nights": "https://images-na.ssl-images-amazon.com/images/I/51Rzum6TgBL._SX331_BO1,204,203,200_.jpg",
+    "sorrows_young_werther": "https://images-na.ssl-images-amazon.com/images/I/51fBHrUsMNL.jpg",
+    "sweet_orange_tree": "https://upload.wikimedia.org/wikipedia/en/thumb/4/4d/My_Sweet_Orange_Tree_2012_poster.jpg/220px-My_Sweet_Orange_Tree_2012_poster.jpg",
+    "what_men_live_by": "https://images-na.ssl-images-amazon.com/images/I/412JkTATtBL._SX331_BO1,204,203,200_.jpg",
+    "the_alchemist": "https://images.gr-assets.com/books/1483412266l/865.jpg"
+  };
+
+
+
+  function printBookDetails(placeId, myObj, cover) {
+    let div = document.getElementById(placeId);
+
+    function size(obj) {
```

Comment:
I liked the `size` function and it's implementing, and I didn't think for a once of counting the object size like so.
But as the next comment says if you get the `Object.keys` till here you can use `booksNames.length`.
if this function was only for practice! Keep the great work :muscle:, but in a separated file or branch, because your homework is our reference let's say on your understanding right?